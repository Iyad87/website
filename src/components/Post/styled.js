import styled from 'styled-components';

import Panel, { panelPadding } from '../Panel';
import Rule from '../Rule';
import { spacing } from '../../styles/spacing';
import { textLight, textLighter } from '../../styles/colours';

export const PostRule = styled(Rule)`
  margin: ${panelPadding} -${panelPadding};
`;

export const Thumbnail = styled.img`
  border-radius: 50%;
  margin-right: ${spacing()};
  height: 46px;
  width: 46px;
`;

export const Title = styled.h4`
  color: ${textLight};
  font-size: 1.375rem;
  font-weight: 400;
  margin-top: 0;
  margin-bottom: ${spacing()};
`;

export const Footer = styled.footer`
  align-items: center;
  display: flex;
  font-size: 1.125rem;
`;

export const Time = styled.time`
  color: ${textLighter};
  display: block;
`;

export const Authored = styled.p`
  margin: 0;
`;

export const Body = styled.section`
  color: ${textLighter};
  font-size: 1.125rem;
`;

export const Article = Panel.withComponent('article');
