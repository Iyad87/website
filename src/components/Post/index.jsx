import React from 'react';
import styled from 'styled-components';

import Panel, { panelPadding } from '../Panel';
import Rule from '../Rule';

const PostRule = styled(Rule)`
  margin: ${panelPadding} -${panelPadding};
`;

const Post = ({ title, body, image, event, date }) => (
  <Panel>
    <PostRule />
  </Panel>
);

export default Post;
