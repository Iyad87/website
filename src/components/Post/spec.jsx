import React from 'react';
import { shallow } from 'enzyme';
import Post from './index';

test('components/Post snapshot', () => {
  const wrapper = shallow(<Post />);

  expect(wrapper).toMatchSnapshot();
});
