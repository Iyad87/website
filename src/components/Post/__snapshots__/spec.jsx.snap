// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Post snapshot 1`] = `
<styled.article>
  <header>
    <styled.h4>
      Weâ€™re Hiring! YLD on the look out for Front-End Devs and DevOps.
    </styled.h4>
  </header>
  <styled.section>
    I think its one of the simplest yet most powerful extensions inVision
    hasn't launched. As animation and transition in inVision evolve it
    should grow even more powerful
  </styled.section>
  <Styled(styled.hr) />
  <styled.footer>
    <styled.img
      src="https://placekitten.com/128/128"
    />
    <styled.p>
      <Link
        href="/people/nuno-job"
      >
        Nuno Job
      </Link>
      <styled.time
        datetime="2018-05-18T14:38:23.996+01:00"
      >
        May 18
      </styled.time>
    </styled.p>
  </styled.footer>
</styled.article>
`;
